### DISCUSSION TITLE
"bad argument" error when running

### DISCUSSION THREAD

**Discussion by terdol** [orbitalquark/textadept#72] [Q&A]
Hello,
I am trying to use Textadept on Linux Lite, but I do not have experience in Lua. I installed the app but when I run it, it always gives the following error:

/core/args.lua:72: bad argument #1 to 'connect' (string expected, got nil)

I tried with 3 different versions including the nightly build and 11.1. Any idea on how I can debug it? Thanks in advance.

**Comment by orbitalquark** [comment]
It would seem that `events.INITIALIZED` is not set up properly. It should be set in *core/events.lua*, which is loaded before *core/args.lua*. Your *~/.textadept/* folder shouldn't be causing any trouble, but just to be safe you can temporarily rename it (or remove it; Textadept will just create another one for you). After unzipping the package, you haven't made any changes to it, right? I really don't know why this would happen out of the box.

**Reply by terdol** [reply]
Thank you for the hint. I think I may have found the problem. In events.lua file, "initialized" word is converted to upper case to get events.INITIALIZED. But when I check in Lua terminal, "initialized" becomes "iNiTiALiZED" as the result of upper() command. Maybe, this is because of my installation, but in python terminal, upper case conversion works correctly. If I can solve this problem, the problem may be resolved.

**Reply by terdol** [reply]
I forgot to write: I tried deleting the ~/.textadept folder, but did not work. And I did not make any change to the sources.

**Comment by orbitalquark** [comment]
This looks like a locale issue. `string.upper()` is doing some locale-dependent uppercasing. I'll have to look into this.

**Reply by terdol** [reply]
Yes. I changed the locale to en_US.UTF-8 and it solved the problem. Thank you.

**Comment by orbitalquark** [comment]
Oh, okay. Great! I'm glad you fixed it :)
